subsys/settings/src/settings_store.c

```diff
@@ -110,6 +109,19 @@ int settings_save(void)
 	}
 	rc = 0;
 
+#if defined(CONFIG_SETTINGS_ROM_HANDLERS)
+	Z_STRUCT_SECTION_FOREACH(settings_handler, ch) {
+		if (ch->h_export) {
+			rc2 = ch->h_export(settings_save_one);
+			if (!rc) {
+				rc = rc2;
+			}
+		}
+	}
+#endif /* CONFIG_SETTINGS_ROM_HANDLERS */
+
+#if defined(CONFIG_SETTINGS_RAM_HANDLERS)
```

Comment:
Ditto, btw you might as well create a helper iterator, e.g. settings_foreach_handler, to make sure that everytime it is iterating in the same order.